### PR TITLE
Remove @Nullable on return types of `elementFromDeclaration` in `TreeUtils`

### PR DESCRIPTION
**Pull request by zcai1**

### CODE REVIEW

**Comment by zcai1** [comment]
It seems whole program inference may scan unprocessed trees?

**Review by wmdietl — CHANGES_REQUESTED** [review]
Thanks for these cleanups, in particular for also removing the now redundant asserts.
There is one failure:
````
> Task :dataflow:checkNullness
/__w/1/s/dataflow/src/main/java/org/checkerframework/dataflow/expression/JavaExpression.java:712: warning: [nullness:unneeded.suppression] warning suppression "nullness:argument" is not used by NullnessChecker
        @SuppressWarnings("nullness:argument") // elementFromDeclaration is non-null for a parameter
````

File: checker/src/main/java/org/checkerframework/checker/guieffect/GuiEffectVisitor.java [hunk 1]
```diff
@@ -405,8 +405,6 @@ public Void visitMethod(MethodTree node, Void p) {
                     "Visiting method " + methElt + " of " + methElt.getEnclosingElement());
         }
```
Comment (wmdietl) [inline checker/src/main/java/org/checkerframework/checker/guieffect/GuiEffectVisitor.java:407]:
```suggestion
        // Check for conflicting (multiple) annotations
```

**Review by wmdietl — APPROVED** [review]
Thanks!

File: checker/src/main/java/org/checkerframework/checker/guieffect/GuiEffectVisitor.java [hunk 2]
```diff
@@ -404,9 +404,7 @@ public Void visitMethod(MethodTree node, Void p) {
             System.err.println(
                     "Visiting method " + methElt + " of " + methElt.getEnclosingElement());
         }
```
Comment (wmdietl) [inline checker/src/main/java/org/checkerframework/checker/guieffect/GuiEffectVisitor.java:406]:
```suggestion
        }

```

**Pull request merged**